src/engine/input/key_mapping/ThriveInputEventWithModifiers.cs

```diff
@@ -0,0 +1,134 @@
+using System;
+using Godot;
+
+public class ThriveInputEventWithModifiers : ICloneable
```

Comment:
I had a really good name suggestion for this, but I can't remember that now. This is a bit long of a name.
Something like `InputKey` or `SpecifiedInputKey` would be better.